src/test/rgw/test_ratelimit_unittest.cc

```diff
@@ -0,0 +1,191 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#include <gtest/gtest.h>
+#include "rgw/rgw_qos.h"
+
+
+using namespace std::chrono_literals;
+
+TEST(RGWRateLimit, shouldratelimit1)
+{
+  // info.enabled = false, so no limit
+  std::unique_ptr<CephContext> cct = std::make_unique<CephContext>(CEPH_ENTITY_TYPE_ANY);
+  std::shared_ptr<ActiveRateLimiter> qos(new ActiveRateLimiter(cct.get()));
+  qos->start();
+  auto active = qos->get_active();
+  RGWQoSInfo info;
+  auto time = ceph::coarse_real_clock::now();
+  std::string key = "uuser123";
+  bool success = active->should_rate_limit("PUT", key, time, &info);
+  EXPECT_EQ(false, success);
+}
+TEST(RGWRateLimit, shouldratelimit2)
+{
+  // check that request is being rejected because there are not enough tokens
+  std::unique_ptr<CephContext> cct = std::make_unique<CephContext>(CEPH_ENTITY_TYPE_ANY);
+  std::shared_ptr<ActiveRateLimiter> qos(new ActiveRateLimiter(cct.get()));
+  qos->start();
+  auto active = qos->get_active();
+  RGWQoSInfo info;
+  info.enabled = true;
+  info.max_read_ops = 1;
+  auto time = ceph::coarse_real_clock::now();
+  std::string key = "uuser123";
+  bool success = active->should_rate_limit("GET", key, time, &info);
+  time = ceph::coarse_real_clock::now();
+  success = active->should_rate_limit("GET", key, time, &info);
+  EXPECT_EQ(true, success);
+}
+TEST(RGWRateLimit, shouldratelimit3)
+{
+  // check that tokens are being filled properly
+  std::unique_ptr<CephContext> cct = std::make_unique<CephContext>(CEPH_ENTITY_TYPE_ANY);
+  std::shared_ptr<ActiveRateLimiter> qos(new ActiveRateLimiter(cct.get()));
+  qos->start();
+  auto active = qos->get_active();
+  RGWQoSInfo info;
+  info.enabled = true;
+  info.max_read_ops = 1;
+  auto time = ceph::coarse_real_clock::now();
+  std::string key = "uuser123";
+  bool success = active->should_rate_limit("GET", key, time, &info);
+  sleep(61);
+  time = ceph::coarse_real_clock::now();
```

Comment:
you can use fake times instead of `sleep()`+`now()`! for example,
```c++
const auto t0 = ceph::coarse_real_clock::now();
...
const auto t1 = t0 + 60s; // check again after 60 seconds
```